stat/roc.go

```diff
@@ -127,3 +129,68 @@ func ROC(cutoffs, y []float64, classes []bool, weights []float64) (tpr, fpr, thr
 
 	return tpr, fpr, cutoffs
 }
+
+// TOC returns the Total Operating Characteristic for the classes provided
+// and the minimum and maximum bounds for the TOC.
+//
+// The input y values that correspond to classes and weights must be sorted
+// in ascending order. classes[i] is the class of value y[i] and weights[i]
+// is the weight of y[i]. SortWeightedLabeled can be used to sort classes
+// together with weights by the rank variable.
+//
+// The returned ntp values can be interpreted as the number of true positives
```

Comment:
I think I can explain why I think this is confusing by re-phrasing it to be more specific: `ntp[i] can be interpreted as the number of true positives where values with ranks greater than (len(classes) - i) are assigned class true for each i from 0 to len(classes).` Optionally, you could also add `So ntp[0] is always zero as this corresponds to assigning all data class false and ntp[len(classes)] is always len(classes) as this corresponds to assigning all data class true.`. It's the `(len(classes) - i)` part that I think is confusing, which is why I would prefer the more specific explanation.

Reply:
I'll see what I can come up with.

Reply:
`ntp[i] = sum_{j = 0}^{i} !classes[j]*weight[j]`

Reply:
I'm going to be obtuse and ask: how many developers out there will recognise the Iverson bracket? We should at least add

```
, where [x] = 1 if x else 0.
```

Reply:
Valid. Done.

Reply:
Thanks.